tests/test_recorders.py

```diff
@@ -0,0 +1,166 @@
+import matplotlib as mpl
+import pytest
+
+mpl.use("Agg")  # headless mode for plt
+import csv
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import genesis as gs
+
+
+@pytest.mark.required
+def test_plotter(tmp_path):
+    """Test if the plotter recorders works."""
+    DT = 0.01
+    STEPS = 10
+
+    scene = gs.Scene(
+        sim_options=gs.options.SimOptions(dt=DT),
+        show_viewer=False,
+        show_FPS=False,
+    )
+
+    scene.add_entity(
+        morph=gs.morphs.Box(size=(0.1, 0.1, 0.1), pos=(0.0, 0.0, 0.5)),
+        material=gs.materials.Rigid(rho=1000.0),
+    )
+
+    call_count = 0
+
+    def dummy_data_func():
+        nonlocal call_count
+        call_count += 1
+        return {
+            "a": [call_count * 0.1, call_count * 0.2, call_count * 0.3],
+            "b": [call_count * 0.01, call_count * 0.02],
+        }
+
+    filepath = tmp_path / "test_plot.mp4"
+    plotter = gs.recorders.MPLPlot(
+        labels={"a": ("x", "y", "z"), "b": ("u", "v")},
+        title="Test MPLPlotter",
+        history_length=50,
+        window_size=(400, 300),
+        hz=1.0 / DT / 2,  # half of the simulation frequency, so every other step
+        show_window=False,
+        save_to_filename=str(filepath),
```

Comment:
You already have a test checking video recording, so you should not check again this feature in this test. Just call `_video_data_func` manually to get a screenshot and do your stuff on it.